packages/apify/src/request_list.ts

```diff
@@ -244,23 +243,21 @@ export class RequestList {
     /**
      * Dictionary of requests that were returned by fetchNextRequest().
      * The key is uniqueKey, value is true.
-     * TODO: Change this to Set
-     * @internal (TODO or should be public maybe?)
+     * @internal
      */
-    inProgress: Record<string, boolean> = {};
+    inProgress = new Set<string>();
 
     /**
      * Dictionary of requests for which reclaimRequest() was called.
      * The key is uniqueKey, value is true.
      * Note that reclaimedRequests is always a subset of inProgress!
```

Comment:
TBH I don't get this `Note that reclaimedRequests is always a subset of inProgress!`.

It seems that `inProgress` should always have all that were ever picked for usage. And then reclaimed duplicates them, if they were scraped at least once. It's kinda weird, but I guess it's because this is not a queue. The index only ever goes up, so requests must never be removed from inProgress until they are handled.

Reply:
Let's remove that, it does not make sense when we have sets. With dicts it was probably true as we just swapped the true/false value, now we literally delete such element from the set.